charts/super-agent/values.yaml

```diff
@@ -0,0 +1,181 @@
+      #   targetPath (Optional): The YAML dot notation path at which the value should be merged. When set, the 
+      #                          valuesKey is expected to be a single flat value. Defaults to `nil` when omitted, which
+      #                          results in the values getting merged at the root.
+
+      # Examples:
+      # - kind: ConfigMap
+      #   name: prod-env-values
+      #   valuesKey: values-prod.yaml
+      # - kind: Secret
+      #   name: prod-tls-values
+      #   valuesKey: crt
+      #   targetPath: tls.crt
+
+    # -- Change the behavior of the operator while installing the chart for the first time. This should only be changed
+    # by advanced users that know what they are doing. Exposes the remediations that the operator is going to try
+    # before give up installing the chart in case it hits an error.
+    # @default -- See `values.yaml`
+    install:
+      crds: CreateReplace  # New CRDs are created, existing CRDs are updated (replaced) but not deleted.
+      replace: true  # Tells the Helm install action to reuse the ‘ReleaseName’, but only if that name is a deleted release which remains in the history.
+      remediation:
+        retries: 3  # Retries is the number of retries that should be attempted on failures before givin up. An uninstall is performed between each attempt.
+
+    # -- Change the behavior of the operator while upgrading the chart. This should only be changed by advanced users
+    # that know what they are doing. Exposes the remediations that the operator is going to try before give up
+    # installing the chart in case it hits an error.
+    # @default -- See `values.yaml`
+    upgrade:
+      crds: CreateReplace  # New CRDs are created, existing CRDs are updated (replaced) but not deleted.
+      force: true  # Force forces resource updates through a replacement strategy.
+      cleanupOnFail: true  # Allows deletion of new resources created during the Helm upgrade action when it fails.
+      remediation:
+        retries: 3  # Retries is the number of retries that should be attempted on failures before givin up. An uninstall is performed between each attempt.
+        strategy: rollback  # rollback or uninstall.
+
+# -- Values for the Flux chat. Ref.: https://github.com/fluxcd-community/helm-charts/blob/flux2-2.10.2/charts/flux2/values.yaml
+# @default -- See `values.yaml`
+flux2:
+  # -- Enable or disable FluxCD installation. New Relic' Super Agent need Flux to work, but the user can use an already
+  # existing Flux deployment. With that use case, the use can disable Flux and use this chart to only install the CRs
+  # to deploy the Super Agent.
+  enabled: true
+
+  # -- The installation of the CRDs is managed by the chart itself.
+  installCRDs: true
+  # -- This is the domain name of the cluster.
+  clusterDomain: cluster.local
+
+  # -- Source controller provides a way to fetch artifacts to the rest of controllers. The source API (which reference
```

Comment:
Is it a good practice in Helm to add comments about components used in other charts? I mean, describing Flux controllers should be the responsibility of our `values.yaml` or the Flux2 one? I was thinking of a way to simplify the values.

Reply:
I defined a simple set of features that might be useful for us and for clients... but the whole values file of the upstream chart is not documented. See:
 * https://github.com/newrelic/helm-charts/pull/1175/files#diff-3bd49393f47bdfcb35b90c33237f384109da730bc892fa317ba8db58d29d7975R84-R85
 * https://github.com/newrelic/helm-charts/pull/1175/files#diff-20bc91206f6699b12924a1284a7942be082f0b582651eedb8f73449236b34c81R23-R26